src/DynamoCoreWpf/Interfaces/IWatchHandler.cs

```diff
@@ -170,8 +172,18 @@ private static string ToString(object obj)
         public WatchViewModel Process(dynamic value, ProtoCore.RuntimeCore runtimeCore, string tag, bool showRawData, WatchHandlerCallback callback)
         {
             return Object.ReferenceEquals(value, null)
-                ? new WatchViewModel(NULL_STRING, tag)
+                ? new WatchViewModel(NULL_STRING, tag, OnRequestSelectGeometry)
                 : ProcessThing(value, runtimeCore, tag, showRawData, callback);
         }
+
+        public event Action<string> RequestSelectGeometry;
+
+        private void OnRequestSelectGeometry(string path)
```

Comment:
It seems, like it's useless. You can create `WatchViewModel` with action `RequestSelectGeometry`. Since in `FindNodeForPath` you check `if (tagGeometry != null)`, 
there is no need to check 2 times `if (RequestSelectGeometry != null)`.

Try remove `OnRequestSelectGeometry` and create `WatchViewModel(value, tag, RequestSelectGeometry);`